app/web_conn.go

```diff
@@ -47,11 +48,18 @@ type WebConn struct {
 	lastAllChannelMembersTime int64
 	lastUserActivityAt        int64
 	send                      chan model.WebSocketMessage
-	sessionToken              atomic.Value
-	session                   atomic.Value
-	connectionID              atomic.Value
-	endWritePump              chan struct{}
-	pumpFinished              chan struct{}
+	// deadQueue behaves like a queue of a finite size
+	// which is used to store all messages that are sent via the websocket.
+	// It basically acts as the user-space socket buffer, and is used
+	// to resuscitate any messages that might have got lost when the connection is broken.
+	// It is implemented by using a circular buffer to keep it fast.
+	deadQueue        []model.WebSocketMessage
+	deadQueuePointer int // Pointer which indicates the next slot to insert.
```

Comment:
Did you name it pointer to tease me or was it how you imagined it? :stuck_out_tongue: 
Interesting that you haven't used the more conventional *tail*.

Reply:
No worries, just being pedantic :) It's clear enough with the comment.